api/server.go.tmpl

```diff
@@ -0,0 +1,168 @@
+/*
+{{ .Copyright }}
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+package api
+
+import (
+	"context"
+	"errors"
+	"math/rand"
+	"net/http"
+	"os"
+	"time"
+
+	"{{ .PkgName }}/common"
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+type server struct {
+	router  *mux.Router
+	version common.Version
+	context context.Context
+	org string
+}
+
+// NewServer creates a new server and starts it
+func NewServer(config common.Config) error {
+	// setup server context with cancellation
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	if config.Org == "" {
+		return errors.New("'org' cannot be empty in the configuration")
+	}
+
+	s := server{
+		// ec2Services:         make(map[string]ec2.EC2),
```

Comment:
Is this left as an example?

Reply:
yeah exactly - i figured i would leave it there